src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -102,14 +102,16 @@ def _compute_mask_indices(
         if sz - min_len <= num_mask:
             min_len = sz - num_mask - 1
 
-        mask_idc = np.random.choice(sz - min_len, num_mask, replace=False)
-        mask_idc = np.asarray([mask_idc[j] + offset for j in range(len(mask_idc)) for offset in range(lengths[j])])
-        mask_idcs.append(np.unique(mask_idc[mask_idc < sz]))
+        mask_idc = torch.randperm(sz - min_len)[:num_mask]
+        mask_idc = torch.from_numpy(
+            np.asarray([mask_idc[j] + offset for j in range(len(mask_idc)) for offset in range(lengths[j])])
+        )
+        mask_idcs.append(torch.unique(mask_idc[mask_idc < sz]))
 
-    min_len = min([len(m) for m in mask_idcs])
+    min_len = torch.min(mask_idcs)
     for i, mask_idc in enumerate(mask_idcs):
```

Comment:
Let's try to get rid of the for-loop and do tensor operations only

Reply:
I am not sure how to do this. Something like following but not getting how should I put it ? Can you help here.

mask[i, mask_idc] = [True, torch.randperm(mask_idc)[:min_len] if torch.tensor(mask_idcs).size() > min_len]